packages/@dcl/inspector/src/lib/sdk/operations/duplicate-entity.ts

```diff
@@ -0,0 +1,37 @@
+import { Entity, IEngine, getComponentEntityTree, Transform } from '@dcl/ecs'
+import { isLastWriteWinComponent } from '../../../hooks/sdk/useComponentValue'
+import updateSelectedEntity from './update-selected-entity'
+
+export function duplicateEntity(engine: IEngine) {
+  return function duplicateEntity(entity: Entity) {
+    const originalToDuplicate: Map<Entity, Entity> = new Map()
+    const TransformComponent = engine.getComponent(Transform.componentId) as typeof Transform
+
+    for (const entityIterator of getComponentEntityTree(engine, entity, TransformComponent)) {
+      const duplicate = engine.addEntity()
+      originalToDuplicate.set(entityIterator, duplicate)
+      // copy values of all components of original entity
+      for (const component of engine.componentsIter()) {
+        if (component.has(entityIterator) && isLastWriteWinComponent(component)) {
+          const componentValue = component.get(entityIterator)
+          component.create(duplicate, JSON.parse(JSON.stringify(componentValue)))
```

Comment:
`JSON.parse(JSON.stringify(componentValue))`
is this necessary ? 🤔

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/structuredClone

Reply:
I tried `structuredClone` but Jest run into an error. I think it's only available in Node 16.